pyuvdata/beamfits.py

```diff
@@ -50,276 +50,276 @@ def read_beamfits(self, filename, run_check=True, check_extra=True,
+                coord_list = ctypes[0:len(ax_names)]
+                if coord_list != ax_names:
+                    raise ValueError('Coordinate axis list does not match coordinate system')
 
             if self.pixel_coordinate_system == 'healpix':
-                basisvec_ax_nums = hxp_basisvec_ax_nums
-                if basisvec_header['CTYPE' + str(basisvec_ax_nums['pixel'])].lower().strip() != 'pix_ind':
-                    raise ValueError('First axis in BASISVEC HDU must be "Pix_Ind" for healpix beams')
-
-                basisvec_Npixels = basisvec_header.pop('NAXIS' + str(basisvec_ax_nums['pixel']))
-
-                if basisvec_Npixels != self.Npixels:
-                    raise ValueError('Number of pixels in BASISVEC HDU does not match '
-                                     'primary HDU')
+                # get pixel values out of HPX_IND extension
+                hpx_hdu = F[hdunames['HPX_INDS']]
+                self.Npixels = hpx_hdu.header['NAXIS2']
+                hpx_data = hpx_hdu.data
+                self.pixel_array = hpx_data['hpx_inds']
+
+                ax_nums = hpx_primary_ax_nums
+                self.nside = primary_header.pop('NSIDE', None)
+                self.ordering = primary_header.pop('ORDERING', None)
+                data_Npixels = primary_header.pop('NAXIS' + str(ax_nums['pixel']))
+                if data_Npixels != self.Npixels:
+                    raise ValueError('Number of pixels in HPX_IND extension does '
+                                     'not match number of pixels in data array')
             else:
-                basisvec_ax_nums = reg_basisvec_ax_nums
-                basisvec_coord_list = [basisvec_header['CTYPE' + str(basisvec_ax_nums['img_ax1'])].lower(),
-                                       basisvec_header['CTYPE' + str(basisvec_ax_nums['img_ax2'])].lower()]
-                basisvec_axis1_array = uvutils.fits_gethduaxis(basisvec_hdu,
-                                                               basisvec_ax_nums['img_ax1'])
-                basisvec_axis2_array = uvutils.fits_gethduaxis(basisvec_hdu,
-                                                               basisvec_ax_nums['img_ax2'])
+                ax_nums = reg_primary_ax_nums
+                self.Naxes1 = primary_header.pop('NAXIS' + str(ax_nums['img_ax1']))
+                self.Naxes2 = primary_header.pop('NAXIS' + str(ax_nums['img_ax2']))
+
+                self.axis1_array = uvutils.fits_gethduaxis(primary_hdu, ax_nums['img_ax1'])
+                self.axis2_array = uvutils.fits_gethduaxis(primary_hdu, ax_nums['img_ax2'])
 
                 # if units aren't defined they are degrees by FITS convention
                 # convert degrees to radians because UVBeam uses radians
-                axis1_units = basisvec_header.pop('CUNIT' + str(basisvec_ax_nums['img_ax1']), 'deg')
+                axis1_units = primary_header.pop('CUNIT' + str(ax_nums['img_ax1']), 'deg')
```

Comment:
Why did 223 conversion require all these variable name changes?  If they're passing the tests, I suppose its fine. But why?

Reply:
I think what happened here is that the code that highlights the individual works got confused by the large diff chunks and showed something misleading. If you look at the diff with whitespace changes filtered out, you'll see that there aren't any variable renames happening here.